### PR TITLE
Add ability to retrieve max for any threshold in faf

### PR DESCRIPTION
**Pull request by mike-w-wilson**
I added on to your function so it works on any threshold in faf.

### CODE REVIEW

File: gnomad/utils/annotations.py [hunk 1]
```diff
@@ -237,73 +237,52 @@ def faf_expr(
...
+    # Iterate through faf thresholds, generally 'faf95' and 'faf99', and
+    # take the highest faf value and it's gen_anc from the sorted faf array
+    for threshold in faf[0].keys():
+        faf_struct = hl.sorted(
+            faf_gen_anc_indices.map(
+                lambda x: {
+                    f"{threshold}_max": hl.or_missing(
+                        faf[x[0]][threshold] > 0, faf[x[0]][threshold]
+                    ),
+                    f"{threshold}_gen_anc": hl.or_missing(
```
Comment (KoalaQin) [inline gnomad/utils/annotations.py:274]: should we just make it `f"{threshold}_max_gen_anc"`, it's long but more accurate.

File: gnomad/utils/annotations.py [hunk 2]
```diff
@@ -237,73 +237,52 @@ def faf_expr(
...
+                        faf[x[0]][threshold] > 0, x[1]["pop"]
+                    ),
+                }
             ),
-        )
+            key=lambda faf: faf[f"{threshold}_max"],
+            reverse=True,
+        )[0]
```
Comment (KoalaQin) [inline gnomad/utils/annotations.py:281]: `[0]` is get the max from the sorted array, right?

**Review by KoalaQin — CHANGES_REQUESTED** [review]
It looks much more elegant, thanks! It's working, I just have two small comments.

**Comment by mike-w-wilson** [comment]
@KoalaQin  -- added but let me know if you think it could be worded more clearly!

File: gnomad/utils/annotations.py [hunk 3]
```diff
@@ -237,73 +237,52 @@ def faf_expr(
...
-                (hl.len(fafs) > 0) & (fafs[0].faf > 0),
-                hl.struct(faf_max=fafs[0].faf, faf_max_gen_anc_grp=fafs[0].population),
-                hl.struct(
-                    faf_max=hl.missing(hl.tfloat),
-                    faf_max_gen_anc_grp=hl.missing(hl.tstr),
-                ),
+    max_fafs_expr = hl.struct()
+
+    # Iterate through faf thresholds, generally 'faf95' and 'faf99', and
+    # take the highest faf value, [0], and it's gen_anc from the sorted faf array
```
Comment (KoalaQin) [inline gnomad/utils/annotations.py:266]:
```suggestion
    # take the highest faf value, '[0]', and its gen_anc from the sorted faf array
   ```

**Review by KoalaQin — CHANGES_REQUESTED** [review]

**Pull request merged**